qiskit_machine_learning/algorithms/serializable_model.py

```diff
@@ -54,7 +54,7 @@ def save(self, *args) -> None:
         """Backwards compatibility with :meth:`to_dill`, deprecated in v0.9.0."""
         issue_deprecation_msg(
             msg="SerializableModelMixin.save() is deprecated.",
-            version="0.9.0",
+            version="0.10.0",
```

Comment:
I believe 0.9.0 was correct here as this comes out in the message that function builds as from when it was deprecated (where in this case it says removal no sooner than 4 months - which presumably would fit fine for a 0.10.0 release where it could be removed)

>     warnings.warn(
>        f"{msg} as of qiskit-machine-learning {version} "
>        f"and will be removed no sooner than {period} after the release date. {remedy}",
>        DeprecationWarning,
>        stacklevel=stacklevel + 1,  # Increment to account for this function.